### ISSUE TITLE
Google search-by-voice displays "no internet connection"

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#2690]
## Test plan
See https://github.com/brave/brave-core/pull/3978

## Description 

Any time you attempt to use voice search on  Brave desktop, it displays a "no internet connection" message. This is after allowing access to your machine's microphone for both the site and Brave itself.
![image](https://user-images.githubusercontent.com/3299098/50357416-cab23800-050a-11e9-9cdc-3033b0ba7745.png)
## Steps to Reproduce

   1. Go to https://google.com/
   2. Click the microphone icon 
![image](https://user-images.githubusercontent.com/3299098/50357473-eb7a8d80-050a-11e9-829b-06c74bc874f1.png)
   3. Attempt to search by voice.

## Actual result:

"No internet connection" is displayed.

## Expected result:
Successful voice search 

## Reproduces how often: 

Always

## Brave version (brave://version info)

v0.58.16

### Reproducible on current release:
- Does it reproduce on brave-browser dev/beta builds? 
Yes, both.

### Website problems only:  
- Does the issue resolve itself when disabling Brave Shields? Negative
- Is the issue reproducible on the latest version of Chrome? Nope!

### Additional Information
Behaves the same on macOS, Windows and Linux systems.

**Comment by LaurenWags** [comment]
Reproduced with 

Brave | 0.60.4 Chromium: 72.0.3626.17 (Official Build) dev (64-bit)
-- | --
Revision | 08d2140079f8db14c9eb885a803d231b1a30e817-refs/branch-heads/3626@{#315}
OS | Mac OS X


Brave | 0.59.12 Chromium: 72.0.3626.17 (Official Build) beta(64-bit)
-- | --
Revision | 08d2140079f8db14c9eb885a803d231b1a30e817-refs/branch-heads/3626@{#315}
OS | Mac OS X

**Comment by RainOfPain125** [comment]
Subbing to this. Can't wait for it to be fixed.

**Comment by btlechowski** [comment]
Reproduced on 

Brave | 0.59.20 Chromium: 72.0.3626.28 (Official Build) beta (64-bit)
-- | --
Revision | 997b1040b63bac324e815797ba52be0cd8f616ed-refs/branch-heads/3626@{#461}
OS | Windows 7

**Comment by Brave-Matt** [comment]
+1 from community:
https://community.brave.com/t/microphone-cannot-connect-to-internet/43382/2

**Comment by johnwaltonvi** [comment]
I have this problem as well.
Going to create a voice command program anyways, but it would be really nice if this bug was fixed. 
Thanks.

**Comment by talkinggoat** [comment]
Same problem, here.

**Comment by Distearth** [comment]
Any update? I'm really ready to switch full time...

**Comment by romeyDog** [comment]
Same issue. Would appreciate any update or workaround. thanks.

**Comment by HennyBro** [comment]
Same issue here on the latest Nightly Build and all other builds from Brave Browser for desktop.

**Comment by HennyBro** [comment]
Seems like they are working on it and expect to fix it "soon"! :D

**Comment by tvking63** [comment]
Just installed Brave yesterday because it's "like Chrome and compatible with all the Chrome stuff" and it took about an hour to find out that the mic doesn't work.  I guess this is like those Linux distros that are "like Windows".  I'm sure there are people working really hard on this but I think the reality is that you're never going to the kind of polished product that you will from a much larger company.

**Comment by rebron** [comment]
@tvking63 It's a bug that we have on our list to fix and given the feedback we can reassess priority. Browsers in general do tend to have long to do lists. In terms of reality, a large company doesn't ensure a polished product and usually trends towards a stagnated/constipated product or one that's bloated because everyone has to get their pet feature in it. My opinion. We are working on polish though and appreciate the feedback. 

We would also appreciate moving non-implementation discussion to community.brave.com though so we can focus on solving the issues here.

**Comment by tvking63** [comment]
Sorry for polluting the implementation discussion.

 I do have another issue that I'm not sure where to go with though because
I don't know how to concisely describe what's going on.  In Settings >
Passwords, The Extensions window, from Settings, about WebTorrent and
Hangouts, is on top of my passwords, part way down the page.  I can't close
or move it and it covers some passwords.  I've attached a screenshot.  If
it's already being worked on can you please point me to a page where I can
monitor the progress.
Thanks

Todd

**Comment by honza-zidek** [comment]
For me it's a showstopper. I cannot use Duolingo so I cannot set Brave as a default browser. And you are aware of this issue since 03.10.2016 - #4476 4476, #3725 3725. I believe it is the same (or connected) issue.

**Comment by honza-zidek** [comment]
> everyone has to get their pet feature in it

This does not sound very promising when you use this phrasing in connection to *non working microphone* in an internet browser in 2019. I was considering starting using Brave but this is very discouraging.

**Comment by rebron** [comment]
@honza-zidek That was a general comment guarding against feature creep and not related to this issue which we've already prioritized to fix. I hope you're encouraged that we do guard against feature creep and bloat and that we do have this issue on our priority list. We will fix this issue and your request to have it done sooner than later has been noted. 

We do need this space now for reproducible test cases and working out the fix.

**Comment by HennyBro** [comment]
@rebron Do not hesitate to update this thread as soon as the first Nighty version supports this! 😊

**Comment by fish2eggs** [comment]
Me too...

Brave | 0.64.77 Chromium: 74.0.3729.169 (Official Build) (64-bit)
Revision | 78e4f8db3ce38f6c26cf56eed7ae9b331fc67ada-refs/branch-heads/3729@{#1013}
OS | Linux

**Comment by rebron** [comment]
cc: @tomlowenthal for api-key

**Comment by tomlowenthal** [comment]
This Google Speech API topic spans multiple issues on this repo. I'm working on it in brave/internal#608. When I resolve things there, I'll come back here with the next technical steps.

**Comment by effrtlss** [comment]
Reproduced with 
Version 0.66.101 Chromium: 75.0.3770.142 (Official Build) (64-bit)
mac OS 10.14

**Comment by effrtlss** [comment]
I noticed other browsers do not even have the google voice option on Google.com . Such as safari, firefox, opera. Maybe easier temp fix would be to comment out voice search code in brave browser until voice search can be enabled?

**Comment by AIDEN44** [comment]
Me too.. 10.11.2019 Any update?
 Version 0.70.123 Chromium: 78.0.3904.97 (Offizieller Build) (64-Bit)

**Comment by npasto** [comment]
Hello I love your browser but for me not being able to use microphone in many websites and plugins is a showstopper ...

Can we please know if you plan to fix this issue in the roadmap ?

 Thanks for your answer because I can't use Brave as my default browser (but I'd like to)  as long as this is not fixed.

 In my case it doesn't work with the 

- https://mybrowseraddon.com/speech-to-text.html extension 
- nor with https://voicenote.in/live/ application. 

Seems like every application trying to use the microphone, **even if  the access permission has been granted** can't use the mic properly ...

Strangely it works with https://www.loom.com/ extension.

I hope this can help, it may be a permission management related stuff ...

**Comment by bsclifton** [comment]
Closing as duplicate of https://github.com/brave/brave-browser/issues/3725

**Comment by bsclifton** [comment]
Re-opening since we have the ability to hide the microphone icon- PR up at https://github.com/brave/brave-core/pull/3978

**Comment by bbondy** [comment]
Sorry this had to be reverted here:
https://github.com/brave/brave-core/pull/4371

Because of test failures that this introduced:
```
[357/357] BraveWalletAPIBrowserTest.DappDetectionWithMetaMaskTestCancel (TIMED OUT)
3 tests timed out:
    BraveWalletAPIBrowserTest.DappDetectionTestAccept (../../brave/browser/extensions/api/brave_wallet_api_browsertest.cc:158)
    BraveWalletAPIBrowserTest.DappDetectionWithMetaMaskTestAccept (../../brave/browser/extensions/api/brave_wallet_api_browsertest.cc:175)
    BraveWalletAPIBrowserTest.DappDetectionWithMetaMaskTestCancel (../../brave/browser/extensions/api/brave_wallet_api_browsertest.cc:191)
```

Re-opening the issue and please re-land when the PR is ready without the CI test failures. Thanks!

**Comment by rishadq** [comment]
As of March 7, 2020 this four-year-old issue is still hanging around. Just updated Brave, but arrived here to see if there was a solution. I realize this is probably a lo-pri issue, but seriously -- it's something that shouldn't face issue starvation for four years.

**Comment by dlxmax** [comment]
This is literally the only problem I have with Brave, and the only reason I keep Chrome around.

**Comment by rebron** [comment]
Looks like this is working now, test in both Windows and macOS 1.8.95.

Brave | 1.8.95 Chromium: 81.0.4044.138 (Official Build) (64-bit)
-- | --
Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS | Windows 10 OS Version 1909 (Build 18363.778)

**Comment by newduke** [comment]
> Same issue. Would appreciate any update or workaround. thanks.

Win-h for windows narrator is my workaround.

**Comment by newduke** [comment]
> Looks like this is working now, tested in both Windows and macOS 1.8.95. Followed STR.
> 
> Brave	1.8.95 Chromium: 81.0.4044.138 (Official Build) (64-bit)
> Revision	8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
> OS	Windows 10 OS Version 1909 (Build 18363.778)

@rebron repro latest Brave, windows 10, shields down (or up)

**Comment by alirzayevme** [comment]
it doesn`t fixed yet 😥

**Comment by tvking63** [comment]
Does not work again as of 70/13/20 and it's been not working for at least a week.  It could be much longer as I don't check it that often. I get a response of "No Internet Connection" It did work for me briefly. Sorry I don't have the dates or version numbers.

Version 1.10.97 Chromium: 83.0.4103.116 (64 Bit)
Windows 10 Home Build 1903 Build 18362.900

**Comment by Slobbering** [comment]
I've updated to the latest Brave browser, Version 1.11.97 Chromium: 84.0.4147.89 (Official Build) (64-bit), but the mic still doesn't work for google search, translate, or on other websites.  The mic also does not work in Microsoft Edge or Opera.

However, Firefox has found a way around it with their own app that works on google.com for searching.
https://addons.mozilla.org/en-US/firefox/addon/voice-fill/
It produces a mic next to the search box.
![image](https://user-images.githubusercontent.com/30689465/87929658-5a3e1500-cab9-11ea-9b15-a389a374e1c0.png)
A bit cumbersome, but it works, slowly.
Any chance of recreating this app in Brave?

**Comment by Wizardling** [comment]
This is still broken.

**Comment by dmiller72** [comment]
Yep. Still broken even with shields down and latest version being used.

**Comment by Wfsneed2** [comment]
Still broken on windows 7. Only feature I need to make Brave my default browser.

**Comment by Zireael07** [comment]
Still broken on Linux Manjaro.

**Comment by adrodanevik** [comment]
Still broken on Kubuntu

**Comment by JoelParke** [comment]
still broken current version on Ubuntu - Version 1.20.103 Chromium: 88.0.4324.152 (Official Build) (64-bit)

**Comment by Realist1001** [comment]
I can confirm Lol, BROKEN

**Comment by ericpardee** [comment]
I too wish search-by-voice would work but it doesn't look like that is going to happen.

TL;DR
Google has built speech recognition on Google sites like Google.com and Google Translate so that they depend on a non-standard API in Google’s web browser which actually just calls back to a Google transcription service.

See this discussion by Brave Support:
https://community.brave.com/t/cant-use-the-microphone-to-search-on-google-com/103175/5

**Comment by Crucizer** [comment]
+1

**Comment by yash10019coder** [comment]
please fix this voice search issue

**Comment by gpcola** [comment]
If the problem is with the Google API then why can't you use an alternative?

**Comment by georgemastro** [comment]
Also a deal-breaker for me on many websites using Google-API for voice search results.

**Comment by Tonev** [comment]
+1 from Brave Community.

https://community.brave.com/t/google-search-by-voice-not-working/277716

**Comment by CryptAlchemy** [comment]
Please fix

**Comment by Wizardling** [comment]
Honestly, it seems unlikely this will ever be sorted :-(

**Comment by arismelachroinos** [comment]
Still broken? I thought Brave was serious

**Comment by Opfour** [comment]
Attempted to do a quick voice search using the latest Brave Version 1.29.81 Chromium: 93.0.4577.82 (Official Build) (64-bit) and I'm still seeing the same issue: "No Internet Connection".

**Comment by justing1216** [comment]
Are you kidding me? This has been an ongoing issue for SIX YEARS NOW?! With no response from an actual Brave Dev or Moderator? Damn @arismelachroinos you're 100% correct, I HAD thought they were serious too. If they can't fix or even bother to reply to a simple issue such as this, imagine the kinds of major issues that they ignore and withhold from us... i.e. ANY SECURITY ISSUE or BREACH in the past SIX FREAKING YEARS. This makes me incredibly nervous and weary of Brave. I will be considering alternatives.
